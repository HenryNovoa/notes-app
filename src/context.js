import React from 'react'

export const NotesContext = React.createContext({
    currentNote: null,
    notes: [
        {id: 1, text: 'Do small assestment'},
        {id: 2, text: 'Go grocery shopping'}
    ]
})

