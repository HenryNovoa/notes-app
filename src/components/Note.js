import { useContext } from 'react'
import { NotesContext } from '../context'

export function Note({ note }) {
    const { dispatch } = useContext(NotesContext)

    return(
        <div id={note.id} className='note'>
            <p>{note.text}</p>
            <div className='btn-container'>
                <button onClick={()=> dispatch({ type: 'SET_CURRENT_NOTE', payload: note})} className='edit'>Edit</button>
            </div>
        </div>
    )
}