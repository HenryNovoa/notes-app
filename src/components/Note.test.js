import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import { NotesContext } from '../context'
import { notesReducer } from '../reducer'
import { Note } from './Note'

jest.mock('../reducer')

describe('Note tests', () => {
  let wrapper = null
  let value = {}
  const wrapperFactory = (wrapperInitialState) => {
    value.state = wrapperInitialState;
    value.dispatch = jest.fn();

    return ({ children }) => (
      <NotesContext.Provider value={value}>
          {children}
      </NotesContext.Provider>
    )
  }

  afterEach(() => {
    jest.restoreAllMocks()
    wrapper = null
  })

  test('Should render loading when loading is true', () => {
    const initialState = {
      pokeReducer: {
        loading: true,
      },
    };
    wrapper = wrapperFactory(initialState);

    render(<PokemonDetail />, { wrapper });

    expect(document.querySelector('.loading-container')).toBeInTheDocument();
  });

  test('Should render error msg when there is an error loading the pokemon', () => {
    const initialState = {
      pokeReducer: {
        error: 'ErrorCode & things',
      },
    };
    wrapper = wrapperFactory(initialState);

    render(<PokemonDetail />, { wrapper });

    expect(document.querySelector('.error__msg').innerHTML)
      .toBe('There has been an error loading the pokemon');
  });

  test('Should render pokemon-detail', () => {
    const initialState = {
      pokeReducer: {
        pokemonDetail: {
          id: 1,
          name: 'bulbasaur',
          types: [{
            type: {
              name: 'grass',
            },
          }],
          sprites: { front_default: 'url', back_default: 'url' },
          stats: [{ stat: { name: 'name' } }],
        },
      },
    };
    wrapper = wrapperFactory(initialState);

    render(<PokemonDetail />, { wrapper });

    expect(document.querySelector('.pokemon-detail')).toBeInTheDocument();
  });

  describe('UseEffect test', () => {});
  test('Should dispatch loadPokemonById', () => {
    const initialState = {
      pokeReducer: {
        pokemonDetail: {
          id: 1,
          name: 'bulbasaur',
          types: [{
            type: {
              name: 'grass',
            },
          }],
          sprites: { front_default: 'url', back_default: 'url' },
          stats: [{ stat: { name: 'name' } }],
        },
      },
    };

    wrapper = wrapperFactory(initialState);

    render(<PokemonDetail />, { wrapper });

    expect(store.dispatch).toHaveBeenCalled();
  });
});