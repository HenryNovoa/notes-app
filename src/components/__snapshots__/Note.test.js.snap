// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note Component expect to render Note component 1`] = `
<div
  className="note"
  id={1}
>
  <p>
    Mock text
  </p>
  <div
    className="btn-container"
  >
    <button
      className="edit"
      onClick={[Function]}
    >
      Edit
    </button>
  </div>
</div>
`;
